_includes/assets/js/indicatorModelHelpers.js

```diff
@@ -0,0 +1,539 @@
+        return selectableFields.every(function(field) {
+          return row[field] === combination[field];
+        });
+      });
+    },
+    getUnitFromStartValues: function(startValues) {
+      var match = startValues.find(function(startValue) {
+        return startValue.field === this.UNIT_COLUMN;
+      }, this);
+      return (match) ? match.value : false;
+    },
+    selectFieldsFromStartValues: function(startValues, selectableFields) {
+      if (!startValues) {
+        return [];
+      }
+      var specialFields = this.nonFieldColumns();
+      var allowedStartValues = startValues.filter(function(startValue) {
+        var normalField = !specialFields.includes(startValue.field);
+        var allowedField = selectableFields.includes(startValue.field)
+        return normalField && allowedField;
+      });
+      var valuesByField = {};
+      allowedStartValues.forEach(function(startValue) {
+        if (!(startValue.field in valuesByField)) {
+          valuesByField[startValue.field] = [];
+        }
+        valuesByField[startValue.field].push(startValue.value);
+      });
+      return Object.keys(valuesByField).map(function(field) {
+        return {
+          field: field,
+          values: valuesByField[field],
+        };
+      });
+    },
+    selectMinimumStartingFields: function(data, fieldNames) {
+      // We filter our full dataset to only those fields.
+      var filteredData = data.filter(function(row) {
+        return fieldNames.some(function(fieldName) {
+          return row[fieldName];
+        });
+      });
+      // We then sort the data by each field. We go in reverse order so that the
+      // first field will be highest "priority" in the sort.
+      fieldNames.reverse().forEach(function(fieldName) {
+        filteredData = filteredData.sort(function(a, b) {
+          return a[fieldName] - b[fieldName];
+        });
+      });
+      // But actually we want the top-priority sort to be the "size" of the
+      // rows. In other words we want the row with the fewest number of fields.
+      filteredData = filteredData.sort(function(a, b) {
+        return Object.keys(a).length - Object.keys(b).length;
+      });
+      // Convert to an array of objects with 'field' and 'values' keys, omitting
+      // any special columns.
+      var specialColumns = this.nonFieldColumns();
+      return Object.keys(filteredData[0]).filter(function(key) {
+        return !specialColumns.includes(key);
+      }).map(function(field) {
+        return {
+          field: field,
+          values: [filteredData[0][field]]
+        };
+      });
+    },
+    nonFieldColumns: function() {
+      return [
+        this.YEAR_COLUMN,
+        this.VALUE_COLUMN,
+        this.UNIT_COLUMN,
+        this.GEOCODE_COLUMN,
+        'Observation status',
+        'Unit multiplier',
+        'Unit measure',
+      ];
+    },
+    getFieldColumnsFromArray: function(arr) {
+      var omit = this.nonFieldColumns();
```

Comment:
Rename `omit` to `columnsToOmit` to make it clearer that it's an array of columns that will later be filtered out.